### ISSUE TITLE
ONNX Export to .view(1, -1) should not use Flatten

### ISSUE THREAD

**Issue by dashesy** [pytorch/pytorch#34390]
## 🐛 Bug

Depending on the input shape `view` could be [exported to](https://github.com/pytorch/pytorch/blob/5500c3de0ac2bec1e79dd7e7f47d9869b70ffb3b/torch/onnx/symbolic_opset9.py#L377) `Flatten`. This is incorrect for some shapes.

## To Reproduce

```python
a = torch.empty((0,)).view(1, -1)
print(a.shape)
```

The output shape is `1, 0` and not `0` nor `0,1`. But if exporting uses `Flatten` the output shape would be `0,1`:

```text
Flatten node: __Flatten_390
Input 0 Name: 822 Shape: {0}
-----------
Output 0 Name: 838 Shape: {0,1}
```

This could be a bug in ORT implementation of Flatten. Reading the specs of [Flatten](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Flatten) however I see `1,0`  will be resulted if `axis=0`. In PyTorch this is used `return g.op("Flatten", self, axis_i=1)` so `axis=1` which may be incorrect.

## Expected behavior

Exporting should preserve correct shapes, same as PyTorch

## Environment

Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
CMake version: version 3.14.0

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 418.87.01
cuDNN version: /usr/local/lib/libcudnn.so.5.1.10

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2019.4                      243
[conda] mkl-service               2.3.0            py36he904b0f_0
[conda] mkl_fft                   1.0.15           py36ha843d7b_0
[conda] mkl_random                1.1.0            py36hd6b4f25_0
[conda] pytorch                   1.4.0           py3.6_cuda10.0.130_cudnn7.6.3_0    pytorch
[conda] torchvision               0.5.0                py36_cu100    pytorch

## Additional context

Export time decision to select between ops is not good to export deterministic models. If shapes are dynamic would be nice to [enforce certain shapes at export time](https://github.com/pytorch/pytorch/issues/34342) different from actual intermediate tensors sha.

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by dashesy** [comment]
Currently I patch like this:

```python
def patch_view_opset9():
    """Patch view to never use Flatten
    Issue: https://github.com/pytorch/pytorch/issues/34390
    """
    import torch.onnx.symbolic_helper as sym_help
    import torch.onnx.symbolic_opset9

    # noinspection PyProtectedMember
    def view(g, self, size):
        size = sym_help._maybe_get_const(size, 'is')
        if sym_help._is_value(size):
            shape = size
        else:
            shape = g.op("Constant", value_t=torch.LongTensor(size))
        return g.op("Reshape", self, shape)

    torch.onnx.symbolic_opset9.view = view
```

**Comment by neginraoof** [comment]
view only uses flatten iff size is a list with lenght 2 and first it's first element matches the first dimension of input.
The repro you provided seems to export successfully in a case like:

```
    class Module(torch.nn.Module):
        def forward(self, x, ):
            x = x.view(1, -1)
            return x

    x = torch.empty((0,))
```
And result is an empty tensor of size: (1, 0)
Are you seeing a failure? Can you provide a repro for that?

**Comment by neginraoof** [comment]
Fix in: https://github.com/pytorch/pytorch/pull/43558

**Comment by garymm** [comment]
Seems fixed by https://github.com/pytorch/pytorch/pull/43558. Please open a new issue if this is still a problem.
